lib/metadataTypes/Automation.js

```diff
@@ -1558,6 +1558,116 @@ class Automation extends MetadataType {
         // delete local copy: retrieve/cred/bu/.../...-doc.md
         await super.postDeleteTasks(customerKey, [`${this.definition.type}-doc.md`]);
     }
+    /**
+     * helper function to retrieve data about all automations in the BU
+     *
+     * @returns {Promise.<string[]>} returns data about automations with the legacy key
+     */
+    static async #getEncodedAutomationIDs() {
+        const keyBackup = this.definition.keyField;
+        const iteratorBackup = this.definition.bodyIteratorField;
+        this.definition.keyField = 'key';
+        this.definition.bodyIteratorField = 'entry';
+        const automationLegacyKeys = await super.retrieveREST(
+            undefined,
+            `/legacy/v1/beta/bulk/automations/automation/definition/`
+        );
+        this.definition.keyField = keyBackup;
+        this.definition.bodyIteratorField = iteratorBackup;
+        return automationLegacyKeys;
+    }
+    /**
+     * A function to update automation email notifications
+     *
+     * @param {string} keys metadata keys
+     * @returns {Promise.<string[]>} keys of the automations where notifications were updated
+     */
+    static async updateNotifications(keys) {
```

Comment:
any publically accessible method on the type classes needs to be added as a placeholder on `metadataTypes/MetadataType.js` as well. Please check how that was done for `MetadataType.execute` and others.

Reply:
looking further, i noticed you did add a method called `setNotifEmailAddr`.
I'm guessing that was an earlier name for `updateNotifications` and only needs to be updated? :)